File: NextSpring/src/main/java/com/study/nextspring/domain/post/post/repository/PostRepository.java
package com.study.nextspring.domain.post.post.repository;

import com.study.nextspring.domain.post.post.entity.Post;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface PostRepository extends JpaRepository<Post,Long>, PostRepositoryCustom {
    List<Post> findByOrderByIdDesc();
}

File: k_rstudy/src/main/kotlin/com/krstudy/kapi/domain/post/repository/PostRepository.kt
package com.krstudy.kapi.domain.post.repository

import com.krstudy.kapi.domain.member.entity.Member
import com.krstudy.kapi.domain.post.entity.Post
import org.hibernate.query.Page

import org.springframework.data.jpa.repository.JpaRepository


interface PostRepository : JpaRepository<Post, Long>, PostRepositoryCustom {
    fun findTop30ByIsPublishedOrderByIdDesc(isPublished: Boolean): List<Post>

}